typespec/Applications.Core/environments.tsp

```diff
@@ -103,24 +110,47 @@ model SecretConfig {
   secret?: string;
 }
 
-@doc("The Cloud providers configuration")
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("ProviderConfigProperties specifies provider configuration details needed for recipes.")
+model ProviderConfigProperties extends Record<unknown> {
+  @doc("The secrets for the referenced provider resource.")
```

Comment:
What do we mean by resource here? This is just configuration.

Could you update the description to include why is it beneficial to use secrets field? @lakshmimsft @ytimocin